src/functional/fencil_processors/runners/gtfn_cpu.py

```diff
@@ -13,36 +13,56 @@
 # SPDX-License-Identifier: GPL-3.0-or-later
 
 
-from typing import Any
+from dataclasses import dataclass, field
+from typing import Any, Optional
 
 import numpy
 
-from functional.fencil_processors.builders import cpp as cpp_callable
-from functional.fencil_processors.codegens.gtfn import gtfn_module as gtfn_codegen
-from functional.fencil_processors.processor_interface import fencil_executor
+from functional.fencil_processors.builders import cache  # , cpp as cpp_callable
+from functional.fencil_processors.builders.cpp import bindings
+from functional.fencil_processors.builders.cpp.build import CMakeProject
+from functional.fencil_processors.codegens.gtfn import gtfn_module
+from functional.fencil_processors.processor_interface import FencilExecutor, FencilProcessorProtocol
+from functional.fencil_processors.source_modules.cpp_gen import CPP_DEFAULT, CppLanguage
 from functional.iterator import ir
 
 
 def convert_arg(arg: Any) -> Any:
     view = numpy.asarray(arg)
     if view.ndim > 0:
-        return memoryview(view)
+        return memoryview(view)  # type: ignore[arg-type] # mypy seems unaware that ndarray is compatible with buffer protocol
     else:
         return arg
```

Comment:
It is this function which is missing something, namely support for arguments other than scalars and located fields (for example constant fields are not supported). That support is explicitly out of scope, however if the tests pass without wrapping arrays in `memoryview`, then I don't mind removing that.

Reply:
Funny story: I commented out the `memoryview` line and suddenly all the builtins tests passed locally for the first time for me. Could have been something different but the change didn't make anything fail, so I am keeping it.

Reply:
Yeah, this is a mess. 

One problem is that numpy will convert literally anything to an array with `asarray`, even scalars. You must check the dimensions afterwards.

As far as I know, the buffer protocol is supported if the object can be converted to a memory view. The other problem is that our `Field` class does not support the buffer protocol, but it does support conversion to numpy arrays, and numpy arrays do support the buffer protocol.

The real fix to this would be to make `Field` support the buffer protocol and do an attempt to cast the argument into a memory view to detect support.

Reply:
How about leaving the mess and waiting for the no-storage implementation? That'll make the problem go away, right?

Reply:
Not at all. The no-storage implementation in functional will only add the feature of allocating buffers with specific memory layouts.

Reply:
Hmmm, it's not exactly correct what I said before, let me rephrase it: the no-storage implementation will not solve this problem in particular but it will require to deal with field buffers in a different and more coherent way.

Reply:
Anyway, the important point of my suggestion is that we should restrict the conversion to ndarray only when the argument defines an `__array__` method, not always.